lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -286,6 +300,76 @@ def safe_create_partitioned_table(table, key:, type:, infer_primary_key: nil, **
     end
   end
 
+  def safe_partman_create_parent(table, key:, interval:, **options)
+    raise ArgumentError, "Expected <key> to be present" unless key.present?
+    raise ArgumentError, "Expected <interval> to be present" unless interval.present?
+
+    invalid_options = options.keys - PARTMAN_CREATE_PARENT_OPTIONS
+
+    raise ArgumentError, "Unrecognized optional argument(s): #{invalid_options}" unless invalid_options.empty?
+
+    if ActiveRecord::Base.connection.postgresql_version < 10_00_00
+      raise PgHaMigrations::InvalidMigrationError, "Native partitioning not supported on Postgres databases before version 10"
+    end
+
+    options[:template_table] = _fully_qualified_name(options[:template_table]) if options[:template_table].present?
+
+    options = options.merge(
+      parent_table: _fully_qualified_name(table),
+      control: key,
+      type: "native",
+      interval: interval,
+    ).compact
+
+    create_parent_sql = options.map { |k, v| "p_#{k} := '#{v}'" }.join(", ")
+
+    connection.execute("SELECT #{_quoted_partman_schema}.create_parent(#{create_parent_sql})")
+  end
+
+  def safe_partman_update_config(table, **options)
```

Comment:
I think this could benefit from explicit keyword args also.

Reply:
I think this one actually has a stronger argument against using keyword args. Since this is updating a row in the db, the presence of an argument with a nil value actually has meaning, and you wouldn't be able to distinguish that with a method signature like:

```ruby
def foo(a: nil, b: nil)
```

We probably wouldn't want to null out columns `b` if the method is called with `foo(a: "some value")`. And if we filter out options with nil values, we wouldn't be able to set a column to null when called with `foo(a: nil)`

Reply:
Oh, hmm, that's a good point.